presto-main/src/main/java/io/prestosql/dispatcher/DispatchInfo.java

```diff
@@ -52,21 +63,42 @@ private DispatchInfo(Optional<CoordinatorLocation> coordinatorLocation, Optional
         this.queuedTime = requireNonNull(queuedTime, "queuedTime is null");
     }
 
+    /**
+     * Returns the coordinator's uri.
+     *
+     * @return an Option {@link CoordinatorLocation}.
+     */
     public Optional<CoordinatorLocation> getCoordinatorLocation()
     {
         return coordinatorLocation;
     }
 
+    /**
+     * If there was a failure, it returns detailed information such as the type, the message
```

Comment:
The summary should be only: "Returns detailed information about a given failure."

Your description could bring the details: "Returns detailed information about an executed failure, such as its type,  message and cause."